source/javascripts/services/request-service.ts

```diff
@@ -0,0 +1,158 @@
+import { getAppSlug } from "./app-service";
+import StringService from "./string-service";
+import { Logger } from "./logger";
+import { StatusType } from "@datadog/browser-logs";
+
+enum RequestServiceMode {
+	Website = "website",
+	Cli = "cli"
+}
+
+interface RequestServiceConfigureOptions {
+	mode?: string;
+	appSlug?: string;
+	logger?: Logger;
+}
+
+class RequestService {
+	public mode = "";
+	public appSlug = "";
+	private logger: Logger | undefined;
+
+	constructor(logger: Logger) {
+		this.configure({ logger });
+	}
+
+	public configure({ mode, appSlug, logger }: RequestServiceConfigureOptions = {}): void {
+		this.mode = mode || (this.modeFromEnvVars() as string);
+		this.appSlug = appSlug || (getAppSlug() as string);
+		this.logger = logger;
+	}
+
+	public async getAppConfigYML(
+		abortedPromise: Promise<undefined>
+	): Promise<string | Error | { bitrise_yml: string; error_message: Error }> {
+		const controller = new AbortController();
+		if (abortedPromise) {
+			abortedPromise.then(() => {
+				controller.abort();
+			});
+		}
+
+		let requestURL = "";
+
+		switch (this.mode) {
```

Comment:
We'll only ever have 2 cases here, I think it's better to user a ternary for this